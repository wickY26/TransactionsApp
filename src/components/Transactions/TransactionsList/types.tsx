import { Transaction } from '../../../containers/Transactions/types';

export interface TransactionsListProps {
  transactions: Transaction[];
};
