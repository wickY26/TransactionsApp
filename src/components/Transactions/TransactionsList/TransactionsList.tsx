import Avatar from '@material-ui/core/Avatar';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemText from '@material-ui/core/ListItemText';
import AccountBalanceIcon from '@material-ui/icons/AccountBalance';
import React, { FC } from 'react';

import { TransactionsListProps } from './types';
import * as SC from './styles/StyledComponents';

const TransactionsList: FC<TransactionsListProps> = ({ transactions }) => (
  <List data-testid="transactions-list" disablePadding>
    {
      transactions.map(({ from, to, amount, description, id }) => (
        <ListItem key={id}>
          <ListItemAvatar>
            <Avatar>
              <AccountBalanceIcon />
            </Avatar>
          </ListItemAvatar>
          <ListItemText secondary={description}>
            <SC.AccountText>{`${from.name} ${from.surname} `}</SC.AccountText>
            sent
            <SC.MoneyText>{` ${amount}â‚¬ ` }</SC.MoneyText>
            to
            <SC.AccountText>{` ${to.name} ${to.surname}`}</SC.AccountText>
          </ListItemText>
        </ListItem>
      ))
    }
  </List>
);

export default TransactionsList;