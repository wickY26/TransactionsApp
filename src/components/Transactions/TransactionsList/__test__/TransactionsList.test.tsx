import React from 'react';
import { render } from '@testing-library/react';

import TransactionsList from '..';

const mockAccount = {name: 'Poyraz', surname: 'Yilmaz', id: 0};
const mockTransactions = [{ from: mockAccount, to: mockAccount, amount: 50, description: 'description', id: 0 }];

const renderComponent = () => render(
  <TransactionsList transactions={mockTransactions} />
);

describe('TransactionsList', () => {

  it('should render correctly', () => {
    const component = renderComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

});
