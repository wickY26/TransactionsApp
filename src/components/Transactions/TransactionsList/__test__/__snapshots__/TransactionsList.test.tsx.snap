// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsList should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 10h3v7H4zM10.5 10h3v7h-3zM2 19h20v3H2zM17 10h3v7h-3zM12 1L2 6v2h20V6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          <p
            class="MuiTypography-root sc-bdfBwQ gKEqEy MuiTypography-body1"
          >
            Poyraz Yilmaz 
          </p>
          sent
          <p
            class="MuiTypography-root sc-gsTCUz dPuDXz MuiTypography-body1"
          >
             50â‚¬ 
          </p>
          to
          <p
            class="MuiTypography-root sc-bdfBwQ gKEqEy MuiTypography-body1"
          >
             Poyraz Yilmaz
          </p>
        </span>
        <p
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          description
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
